lib/Test/Script.pm

```diff
@@ -313,6 +315,21 @@ sub script_runs {
   goto &_run;
 }
 
+=head2 script_fails
+
+L</script_runs> may be invoked as L</script_fails>. The exit option is mandatory when used this way. Since Perl 5.12, C<die> usually returns 255, but does not promise to do so. Fatal errors like divide by 0 also return 255 often so it is not the best error code for a trapped exception. L<script_runs> needs an exit code it considers success, use C<warn; exit;> instead of die.
+
+=cut
+
+sub script_fails {
+  my $args   = _script(shift);
+  my ( $opt, $testname ) = @_;
+  die "exit is a mandatory option for script_fails"
+    unless eval{ defined $opt->{exit} };
+  $testname = "Script $args->[0] fails" unless defined $testname;
+  script_runs( $args, $opt, $testname );
```

Comment:
This will report failed tests here instead of the `script_fails` caller, which is what we want.

```suggestion
  my $context = context();
  return release $context, script_runs( $args, $opt, $testname );
```

I haven't actually tried the release function before, but it is documented here:  https://metacpan.org/pod/Test2::API#release($;$)